internal/httpserve/middleware/auth/authoptions.go

```diff
@@ -0,0 +1,393 @@
+// using this option, the defaults will still be preserved if not set on the input.
+func WithAuthOptions(opts AuthOptions) AuthOption {
+	return func(conf *AuthOptions) {
+		if opts.KeysURL != "" {
+			conf.KeysURL = opts.KeysURL
+		}
+
+		if opts.Audience != "" {
+			conf.Audience = opts.Audience
+		}
+
+		if opts.Issuer != "" {
+			conf.Issuer = opts.Issuer
+		}
+
+		if opts.MinRefreshInterval != 0 {
+			conf.MinRefreshInterval = opts.MinRefreshInterval
+		}
+
+		if opts.Context != nil {
+			conf.Context = opts.Context
+		}
+	}
+}
+
+// WithJWKSEndpoint allows the user to specify an alternative endpoint to fetch the JWKS
+// public keys from. This is useful for testing or for different environments.
+func WithJWKSEndpoint(url string) AuthOption {
+	return func(opts *AuthOptions) {
+		opts.KeysURL = url
+	}
+}
+
+// WithAudience allows the user to specify an alternative audience.
+func WithAudience(audience string) AuthOption {
+	return func(opts *AuthOptions) {
+		opts.Audience = audience
+	}
+}
+
+// WithIssuer allows the user to specify an alternative issuer.
+func WithIssuer(issuer string) AuthOption {
+	return func(opts *AuthOptions) {
+		opts.Issuer = issuer
+	}
+}
+
+// WithMinRefreshInterval allows the user to specify an alternative minimum duration
+// between cache refreshes to control refresh behavior for the JWKS public keys.
+func WithMinRefreshInterval(interval time.Duration) AuthOption {
+	return func(opts *AuthOptions) {
+		opts.MinRefreshInterval = interval
+	}
+}
+
+// WithContext allows the user to specify an external, cancelable context to control
+// the background refresh behavior of the JWKS cache.
+func WithContext(ctx context.Context) AuthOption {
+	return func(opts *AuthOptions) {
+		opts.Context = ctx
+	}
+}
+
+// WithValidator allows the user to specify an alternative validator to the auth
+// middleware. This is particularly useful for testing authentication.
+func WithValidator(validator tokens.Validator) AuthOption {
+	return func(opts *AuthOptions) {
+		opts.validator = validator
+	}
+}
+
+// WithReauthenticator allows the user to specify a reauthenticator to the auth
+// middleware.
+func WithReauthenticator(reauth Reauthenticator) AuthOption {
+	return func(opts *AuthOptions) {
+		opts.reauth = reauth
+	}
+}
+
+// Reauthenticate is a middleware helper that can use refresh tokens in the echo context
+// to obtain a new access token. If it is unable to obtain a new valid access token,
+// then an error is returned and processing should stop.
+func Reauthenticate(conf AuthOptions, validator tokens.Validator) func(c *echo.Context) (string, error) {
+	// If no reauthenticator is available on the configuration, always return an error.
+	if conf.reauth == nil {
+		return func(c *echo.Context) (string, error) {
+			return "", ErrRefreshDisabled
+		}
+	}
+
+	// If the reauthenticator is available, return a function that utilizes it.
+	return func(c *echo.Context) (_ string, err error) {
```

Comment:
```suggestion
	return func(c *echo.Context) (string,  error) {
```

This is odd to have one named var on the return and one not